torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -286,6 +287,48 @@ RegisterOperators reg({
             return 0;
           };
         }),
+    Operator(
+        prim::BroadcastSizes,
+        [](const Node* node) -> Operation {
+          size_t num_inputs = node->inputs().size();
+          return [num_inputs](Stack& stack) {
+            std::vector<int64_t> size;
+            size.reserve(8);
+            for (size_t i = 0; i < num_inputs; ++i) {
+              size = at::infer_size(size, peek(stack, i, num_inputs).toIntList()->elements());
+            }
+            drop(stack, num_inputs);
+            push(stack, std::move(size));
+            return 0;
+          };
+        }),
+    Operator(
+        prim::ChunkSizes,
```

Comment:
(observation) This feels a lot like complete shape analysis

Reply:
Because it's exactly that, except that it's shifted to run-time, because we don't have those things available statically.